NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h

```diff
@@ -835,16 +835,16 @@ class NEOMATHENGINE_API IDnnEngine : public IBlasEngine {
 		const CBlobDesc& result ) = 0;
 
 	virtual void BlobMeanPooling( const CMeanPoolingDesc& desc, const CFloatHandle& source, const CFloatHandle& result ) = 0;
-	virtual void BlobMeanPoolingBackward( const CMeanPoolingDesc& desc, const CFloatHandle& outputDiff, const CFloatHandle& inputDiff ) = 0;
+	virtual void BlobMeanPoolingBackward( const CMeanPoolingDesc& desc, const CFloatHandle& resultDiff, const CFloatHandle& sourceDiff ) = 0;
 
 	// GlobalMaxOverTime-Pooling
 	// The descriptor should be destroyed using the standard delete operator after use.
 	virtual CGlobalMaxOverTimePoolingDesc* InitGlobalMaxOverTimePooling( const CBlobDesc& source, const CBlobDesc& result ) = 0;
 
 	virtual void BlobGlobalMaxOverTimePooling( const CGlobalMaxOverTimePoolingDesc& desc, const CFloatHandle& source, const CIntHandle* maxIndices,
 		const CFloatHandle& result ) = 0;
-	virtual void BlobGlobalMaxOverTimePoolingBackward( const CGlobalMaxOverTimePoolingDesc& desc, const CFloatHandle& source, const CIntHandle& maxIndices,
-		const CFloatHandle& result ) = 0;
+	virtual void BlobGlobalMaxOverTimePoolingBackward( const CGlobalMaxOverTimePoolingDesc& desc, const CFloatHandle& sourceDiff, const CIntHandle& maxIndices,
```

Comment:
Wrong names! See `CMaxOverTimePoolingLayer::BackwardOnce`. Also fix all the `IMathEngine` implementations you've changed

Reply:
Do I understand correct, that in the original signature the arguments' names were reversed
```
MathEngine().BlobMaxOverTimePoolingBackward( *desc, outputDiffBlobs[0]->GetData(),
			maxIndices->GetData<int>(), inputDiffBlobs[0]->GetData() );
```
So should be correct
```
void BlobGlobalMaxOverTimePoolingBackward( const CGlobalMaxOverTimePoolingDesc& desc,
			const CFloatHandle& resultDiff, const CIntHandle& maxIndices, const CFloatHandle& sourceDiff ) = 0;
```

Reply:
Yep. Yet another backward function calls sourceDiff based on resultDiff.